src/FieldEncryptMapBase.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\pants\FieldEncryptMapBase.
+ */
+
+namespace Drupal\field_encrypt;
+
+use Drupal\Core\Plugin\ContextAwarePluginBase;
+
+/**
+ * This class structure mirrors the way blocks are constructed.
+ * We have a base class and a plugin interface.
+ *
+ * @package Drupal\field_encrypt
+ */
+abstract class FieldEncryptMapBase extends ContextAwarePluginBase implements FieldEncryptMapPluginInterface/*, ContainerInjectionInterface*/ {
+
+  protected $container;
+
+  public function __construct($configuration, $plugin_id, $plugin_definitions) {
+    parent::__construct($configuration, $plugin_id, $plugin_definitions);
+    $this->container = \Drupal::getContainer();
```

Comment:
@nerdstein please let me know if there is a better way to do this in plugins. I couldn't determine how to load services / get the container without a call to this directly. I understand the pattern for DI in services and controllers, but not plugins.